### ISSUE TITLE
감정 분석 모델이 짧은 비속어 단독 입력에서 감정을 다르게 판정합니다 (예: "등신"→긍정, "멍청이"→부정)

### ISSUE THREAD

**Issue by ndopy** [HANCOM-E/pulse-frontend#333]
## 배경

최종 발표 시연 리허설을 준비하면서, 모더레이션 큐가 채워지는 장면을 보여주려고 소감 제출 화면에 "등신"과 "멍청이" 두 단어를 각각 제출해봤습니다.  
두 단어 모두 독성(toxic)으로는 정상적으로 잡혀 모더레이션 큐에 들어갔습니다.  
다만 감정(sentiment) 판정은 "등신"은 긍정(POS), "멍청이"는 부정(NEG)으로 서로 다르게 나왔습니다.

<img width="1711" height="908" alt="Image" src="https://github.com/user-attachments/assets/402d53f7-ef8d-44db-a993-9e336010fd01" />

### 독성(toxic) 판정은 정상적으로 됩니다

두 단어 모두 `lib/tagger/tagger.js`의 `PROFANITY` 목록(AI 모델과 무관하게 코드가 직접 문자열을 대조하는 욕설 단어 목록)에 있습니다.  
그래서 제출 즉시 `toxic` 값이 `true`로 잡힙니다.  
주최자 대시보드의 모더레이션 큐(`ModerationQueue`)에 두 소감 모두 정상적으로 나타납니다.

### 감정(sentiment) 판정이 두 단어에서 다르게 나옵니다

반면 감정 판정은 두 단어가 서로 다른 결과를 냅니다.

- **"등신"** 은 긍정(POS)으로 분류됩니다.
- **"멍청이"** 는 부정(NEG)으로 분류됩니다.

## 원인

**두 단어의 감정 판정은 규칙이 아니라 AI 모델의 로짓(logits, 모델이 분류 직전에 내놓는 원시 점수) 값만으로 정해졌습니다.**  
이렇게 되는 이유는 `lib/tagger/tagger.js`의 `ruleLabel` 함수가 두 단어 모두에 대해 판정을 AI 모델로 넘겨버리기 때문입니다.  
`ruleLabel`은 문장에 `POSITIVE_WORDS`·`NEGATIVE_WORDS` 목록에 있는 단어가 있거나 질문형 문장일 때만 감정을 직접 확정합니다.  
그 외에는 `null`을 반환해 판정을 AI 모델에 넘깁니다.  
"등신"과 "멍청이" 둘 다 이 두 목록 어디에도 없고 질문형도 아니라서, `ruleLabel`은 두 단어 모두에 대해 `null`을 반환합니다.  
그래서 두 단어의 감정 라벨은 `labelFromMargin(marginOf(logits), tau)` 한 줄, 즉 AI 모델이 계산한 로짓 값과 `tau`(중립 판정 임계값, 기본값 1.2)만으로 정해집니다.

이 AI 모델(`koelectra-small-v3-nsmc`, `TAGGER_VERSION` 상수에 명시)은 NSMC(네이버 영화 리뷰 말뭉치)로 학습된 소형 한국어 감정 분류 모델입니다.  
이 모델은 "영화가 재미없었어요"처럼 문장 단위 리뷰를 학습했습니다.  
"등신"·"멍청이"처럼 맥락 없는 단어 하나만 있는 입력은 이 학습 데이터에 거의 없었을 가능성이 높습니다.  
그래서 이런 짧은 비속어 단독 입력에서는 모델이 안정적인 판단을 못 합니다.  
사람이 보기엔 비슷한 수준의 욕설이더라도, 모델 내부 계산상으로는 반대 방향으로 갈릴 수 있습니다.

### 지금 화면에 미치는 영향은 없습니다

두 단어 모두 `toxic`이 `true`라 `status`가 `HIDDEN`으로 저장됩니다.  
대시보드의 감정 분포·긍정 비율 계산은 `VISIBLE` 상태인 소감만 봅니다.  
그래서 이번에 확인한 두 단어의 감정 오분류 자체는 지금 화면 어디에도 드러나지 않습니다.  
다만 이 현상은 짧고 맥락 없는 입력 전반에서 모델이 불안정할 수 있다는 신호입니다.  
욕설 목록에 없는 다른 짧은 표현("헐"·"음" 같은 단어)에서도 비슷한 오분류가 날 수 있다고 보고 기록해둡니다.

### 실제 로짓 수치는 확인하지 못했습니다

`scripts/tagger-eval.mjs`와 같은 방식으로 이 두 단어를 로컬에서 직접 모델에 넣어 실제 margin 값을 확인해보려 했습니다.  
이 기기에 설치된 `onnxruntime-node` 네이티브 바이너리가 로드되지 않는 환경 문제(`Cannot find module '../bin/napi-v6/darwin/x64/onnxruntime_binding.node'`)로 실행이 되지 않았습니다.  
그래서 정확한 margin 수치까지는 확인하지 못했습니다.  
태거 로직 자체와는 무관한 로컬 환경 문제입니다.

### 지금 당장 조치가 필요한 이슈는 아닙니다

코드를 고쳐야 하는 버그라기보다 AI 모델의 알려진 한계를 기록해두는 목적입니다.  
그래서 라벨을 `question`으로 지정했습니다.  
다만 팀이 원하면, `ruleLabel`의 `NEGATIVE_WORDS` 목록(`PROFANITY` 목록과는 별개입니다)에 비속어를 추가할 수 있습니다.  
그러면 모델 결과와 무관하게 강제로 `NEG`로 고정할 수 있습니다.
